kornia/morphology/morphology.py

```diff
@@ -1,29 +1,81 @@
 import torch
-import torch.nn as nn
 import torch.nn.functional as F
 
-from kornia.morphology.basic_operators import dilation, erosion
-from kornia.morphology.open_close import open, close
+from typing import List, Optional
 
 
-# morphological gradient
-def gradient(tensor: torch.Tensor, kernel: torch.Tensor) -> torch.Tensor:
-    r"""Returns the morphological gradient of an image.
+# Dilation
+def dilation(tensor: torch.Tensor, kernel: torch.Tensor, origin: Optional[List[int]] = None) -> torch.Tensor:
+    r"""Returns the dilated image applying the same kernel in each channel.
 
-    That means, (dilation - erosion) applying the same kernel in each channel.
-    The kernel must have 2 dimensions, each one defined by an odd number.
+    The kernel must have 2 dimensions.
 
     Args:
-       tensor (torch.Tensor): Image with shape :math:`(B, C, H, W)`.
-       kernel (torch.Tensor): Structuring element with shape :math:`(k_x, k_y)`.
+        tensor (torch.Tensor): Image with shape :math:`(B, C, H, W)`.
+        kernel (torch.Tensor): Structuring element with shape :math:`(k_x, k_y)`.
+        origin (List[int], Tuple[int, int]): Origin of the structuring element. Default is None and uses the center of
+        the structuring element as origin (rounding towards zero).
 
     Returns:
-       torch.Tensor: Dilated image with shape :math:`(B, C, H, W)`.
+        torch.Tensor: Dilated image with shape :math:`(B, C, H, W)`.
 
     Example:
         >>> tensor = torch.rand(1, 3, 5, 5)
         >>> kernel = torch.ones(3, 3)
-        >>> gradient_img = gradient(tensor, kernel)
+        >>> dilated_img = dilation(tensor, kernel)
+    """
+
+    if not isinstance(tensor, torch.Tensor):
+        raise TypeError("Input type is not a torch.Tensor. Got {}".format(
+            type(tensor)))
+
+    if len(tensor.shape) != 4:
+        raise ValueError("Input size must have 4 dimensions. Got {}".format(
+            tensor.dim()))
+
+    if not isinstance(kernel, torch.Tensor):
+        raise TypeError("Kernel type is not a torch.Tensor. Got {}".format(
+            type(kernel)))
+
+    if len(kernel.shape) != 2:
+        raise ValueError("Kernel size must have 2 dimensions. Got {}".format(
+            kernel.dim()))
+
+    # origin
+    se_h, se_w = kernel.shape
+    if origin is None:
+        origin = [se_h // 2, se_w // 2]
+
+    # pad
+    pad_e: List[int] = [origin[1], se_w - origin[1] - 1, origin[0], se_h - origin[0] - 1]
+    output: torch.Tensor = F.pad(tensor, pad_e, mode='constant', value=0.)
+
+    # computation
+    output = output.unfold(2, se_h, 1).unfold(3, se_w, 1)
+    output = torch.amax(output + kernel.flip((0, 1)), (4, 5))
```

Comment:
This makes jit CUDA tests fail. Could you please use torch.max for a while and when JIT will support amax, then we will use ama.

Reply:
My jit CUDA tests pass with amax, but I can put max if yours don't

Reply:
Sure! I saw that. I am refactoring it.

Reply:
I changed it! Does pass the test now?